From 74bca42e1613c58805d7b048841c2fa8f8502158 Mon Sep 17 00:00:00 2001
From: "Tuan M. Hoang" <tmhoang@flatglobe.org>
Date: Tue, 14 Mar 2017 16:44:04 -0400
Subject: [PATCH] s390x: fix fpreg_t and remove unused per_struct

Including sys/procfs.h complains unknown type name 'fpreg_t' in
bits/user.h. fpreg_t in bits/signal.h and elf_fpreg_t in bits/user.h
are practically the same.

per_struct is never used, even conflicts with kernel header
asm/ptrace.h
---
 arch/s390x/bits/user.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/arch/s390x/bits/user.h b/arch/s390x/bits/user.h
index 90f07b78..17bce16f 100644
--- a/arch/s390x/bits/user.h
+++ b/arch/s390x/bits/user.h
@@ -8,7 +8,7 @@ typedef union {
 
 typedef struct {
 	unsigned fpc;
-	fpreg_t fprs[16];
+	elf_fpreg_t fprs[16];
 } elf_fpregset_t;
 
 #define ELF_NGREG 27
@@ -32,7 +32,7 @@ struct _user_per_struct {
 	unsigned short perc_atmid;
 	unsigned long address;
 	unsigned char access_id;
-} per_struct;
+};
 
 struct _user_regs_struct {
 	struct _user_psw_struct psw;
-- 
2.11.1

